documentation/homework/week4/src/stores/todo_actions.js

```diff
@@ -0,0 +1,150 @@
+import { observable, action, runInAction } from 'mobx'
+
+// Setting our localStorage
+// import { locals } from '../utils'
+
+// set the default state of the items Object
+const default_item_state = {
+  description: '',
+  deadline: '',
+}
+
+async function request(url, method, inputToString) {
+  const params = {
+    "body": JSON.stringify(inputToString),
+    "headers": {
+      "Content-type": "application/json"
+    }
+  }
+
+  let options = {}
+  method = method ? method.toUpperCase() : "GET"
+  switch (method) {
+    case "POST":
+    case "PATCH":
+      options = { method, ...params, }
+      break
+    default:
+      options = { method }
+  }
+
+  // is't Okay to catch the errors here with in 
+  // -- one single line with the.json line ?
+  // OR we have to catch it on the fetch it self also ?!
+  // Because what i see here is a strange look to me
+  const preResponse = await fetch(`https://hyf-react-api.herokuapp.com/${url}`, options)
+  const response = await preResponse.json().catch(console.error)
+  return response
+}
+
+class todo_actions {
+  @observable // this is only for the adding item
+  item_state = { ...default_item_state }
+
+  @observable // this is only for the editing item
+  item_edit_state = { ...default_item_state }
+
+  // the items By default ~> []
+  @observable
+  items = /* locals.load || */[] // Items Holder ~> Uncomment if there is a localSorage
+
+  @action
+  loadData = async () => {
+    const data = await request('todos')
+    runInAction(() => {
+      this.items = data
+    })
+  }
+
+  @action // onInput Change
+  onInputChange = (value, field) => {
+    this.item_state = {
+      ...this.item_state,
```

Comment:
Why set the entire object? Why not use square bracket syntax? Eg:

    this.item_state[field] = value

Reply:
thank you! i didin't notice that 😅